client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
```

Comment:
I have no concern with this growing in size whatsoever.
What concerns me here (and it's the same discussion I've been having with @dmsnell over the connection middleware) is the separation of concerns.
In my opinion, `middleware.js` should not be concerned with translations/logic specific to some actions.
I do not like this centralized mega-case statements or huge arrays.
For the same reason, I do not like "stylesheet" and "javascript' folder separation in some projects, because I think all component logic should be close to that component. When I'm adding / editing component, I should not be venturing outside that folder too often.

Granted, we did not settle upon correct idea to link these "middleware plugins" with main middleware - as I said, I try to come up with something in the connection middleware efforts.